DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/mmhost/part2_single_host/src/mmhost.cpp

```diff
@@ -0,0 +1,88 @@
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/sycl.hpp>
+
+#include "exception_handler.hpp"
+
+struct SingleMMIP {
+  // This kernel has 3 annotated pointers, but since they have no properties
+  // specified, this kernel will result in the same IP component as Example 1.
+  sycl::ext::oneapi::experimental::annotated_arg<int *> x;
+  sycl::ext::oneapi::experimental::annotated_arg<int *> y;
+  sycl::ext::oneapi::experimental::annotated_arg<int *> z;
+  int size;
+
+  void operator()() const {
+    for (int i = 0; i < size; ++i) {
+      z[i] = x[i] + y[i];
+    }
+  }
+};
+
+int main(void) {
+#if FPGA_SIMULATOR
+  auto selector = sycl::ext::intel::fpga_simulator_selector_v;
+#elif FPGA_HARDWARE
+  auto selector = sycl::ext::intel::fpga_selector_v;
+#else  // #if FPGA_EMULATOR
+  auto selector = sycl::ext::intel::fpga_emulator_selector_v;
+#endif
+
+  bool passed = true;
+
+  try {
+    // create the device queue
+    sycl::queue q(selector, fpga_tools::exception_handler,
+                  sycl::property::queue::enable_profiling{});
+
+    // Print out the device information.
+    sycl::device device = q.get_device();
+    std::cout << "Running on device: "
+              << q.get_device().get_info<sycl::info::device::name>().c_str()
+              << std::endl;
+
+    // Create and initialize the host arrays
+    constexpr int kN = 8;
+    std::cout << "Elements in vector : " << kN << "\n";
+
+    // Here we may use auto* or int* when declaring the pointer interface
+    auto *array_A = sycl::malloc_shared<int>(kN, q);
+    auto *array_B = sycl::malloc_shared<int>(kN, q);
+    int *array_C = sycl::malloc_shared<int>(kN, q);
```

Comment:
Make sure to do this for all allocation calls across all files in this tutorial. I missed this earlier, but make sure to add these:
```suggestion
    int *array_C = sycl::malloc_shared<int>(kN, q);
    assert(array_A);
    assert(array_B);
    assert(array_C);
```
It is strongly recommended that users check the pointer returned by malloc* calls before using them.